### ISSUE TITLE
The first audio file attached to an audio work should be it's representative

### ISSUE THREAD

**Issue by mark-dce** [curationexperts/goldenseal#214]
When I upload the first wav or mp3 file to an audio work, that file should be used as the representative for playback.

**Comment by jcoyne** [comment]
This could cause the importer to behave slowly if there are a lot of files attached and none of them are an wav or mp3 file. Are you sure?  Is there some reason you'd exclude OGG, AIFF, FLAC, AAC, WMA, etc. from the list of preferred formats??

**Comment by mark-dce** [comment]
I meant this story in the context of the UI for browser based upload.  How would the importer know which audio to import anyway, it should be looking for a TEI transcript and the transcript should point at the proper audio file.

I didn't mean to exclude other audio formats, I was just giving the two I listed as the most common examples.  

**REFINED REQUIREMENT**
"When I upload the first playable audio file to an audio work via the UI, it should be set as the representative media."

**Comment by jcoyne** [comment]
The code for the UI is no different than the code used by the importer.  It will impose a performance penalty making the import slower.

**Comment by mark-dce** [comment]
If that's the case, then I don't understand why we said that we could do this for videos (which I thought it was working at one point) - see #63, or why you created #134 which says to do exactly the same thing for videos.

**Comment by val99erie** [comment]
In all the previous cases, such as #63 or #134, we blindly set the first attached file as the representative (without checking its mime-type or any other validations).  We currently don't have any code that checks mime type before selecting a representative.

One reason this story might be difficult is that normally the correct mime type might not be known until after the characterization jobs run.

**Comment by mark-dce** [comment]
Ok, this is sounding hard, so let's back up a step and think about the end-user need I was trying to address and figure out if there are any simple ways to address it.  

**Use Cases**
As a user when I'm using the UI to create works, most cases I'm trying to create simple works that match one of two patterns:
1. they have a single file attached that corresponds to the work "type" - i.e. I'm attaching a tiff to an image work, an mp3 to an audio work, a pdf to a document work
2. the work includes a TEI transcript that specifies the corresponding audio, video, or first image.

**Problems**
In the first case, it feels like a clunky extra step to have to go in after I create the work and attach a file to then say "use that file for this work" if there's one and only one file attached to the work.

In the second case, the transcript actually points to the primary media file or the image for the first page.  Again it feels like making the user do extra work to have to go in and set the representative since it's already specified in the TEI.

**Solutions?**
Are there any other ways we could address this that would avoid the second step of having to edit the work every time I create work via the UI?

Possible options I can see:
- Always use the first file uploaded (even when I create the work first with not files attached and then use the Javascript Uploader) as the representative (even if the file mime type doesn't match the work type).  And just tell the user they have to upload the thing they want as the default representative first (if there are multiple attachments and they want something else they can always edit the work).  If we changed the way we conceived of representative media, we could even use TEI as representative media - if the TEI is a synchronized transcript (contains a <media> node etc.), show AblePlayer - if the TEI is a book (contains a <pb facs="..."> node, etc.), show the book viewer.
- Always use the file selected using the "choose file" button on the new work form as the representative.  (and possibly add a button on Text, Audio, and Video new work form that lets the user choose a transcript as well).  With this option, if I want a one-step process to create works, I'd have to attach them on the new work form, which is OK.
- Give a way to indicate "set this as the representative" at the time the user chooses / uploads the file -it seems like this could be hard with the javascript uploader, but easy with the form based uploader???
- Add behavior to the derivative generator that checks the work the file is being attached to and sets the representative or transcript when appropriate (I can even live with the cases where there are possible race conditions - e.g. where the second audio file gets set as the representative instead of the first audio attached)
- _Other more simple fixes I'm not thinking of ???_
